test/adapters/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2023 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+function(add_adapter_test name)
+    cmake_parse_arguments(args
+        ""                      # options
+        "FIXTURE"               # one value keywords
+        "SOURCES;ENVIRONMENT"   # multi value keywords
+        ${ARGN})
+
+    set(target test-adapter-${name})
+    add_ur_executable(${target} ${args_SOURCES}
+        ${PROJECT_SOURCE_DIR}/test/conformance/source/environment.cpp
+        ${PROJECT_SOURCE_DIR}/test/conformance/source/main.cpp
+    )
+
+    set(fixtures "PLATFORM;DEVICES;KERNELS")
+    if(NOT args_FIXTURE IN_LIST fixtures)
+        message(FATAL_ERROR
+            "FIXTURE must be one of: ${fixtures}. Found: ${args_FIXTURE}")
+    endif()
+
+    target_compile_definitions(${target} PRIVATE
+        ${args_FIXTURE}_ENVIRONMENT)
+    target_link_libraries(${target} PRIVATE
+        ${PROJECT_NAME}::loader
+        ${PROJECT_NAME}::headers
+        ${PROJECT_NAME}::testing
+        ${PROJECT_NAME}::common
+        GTest::gtest)
+
+    add_test(NAME ${target} COMMAND $<TARGET_FILE:${target}>)
+    set_tests_properties(${target} PROPERTIES
+        LABELS "adapter:${name}"
```

Comment:
With this change the adapter tests no longer run in CI. I think this needs to be `LABELS "adapter;${name}"`.
Also, this removes the match file machinery from these tests, but it appears it was never necessary for these particular tests (or I can't find the corresponding match files).

Reply:
> With this change the adapter tests no longer run in CI. I think this needs to be LABELS "adapter;${name}".

Good spot, I'll fix this.

> Also, this removes the match file machinery from these tests, but it appears it was never necessary for these particular tests (or I can't find the corresponding match files).

These should aways pass and since they are adapter specific by nature we don't need the ability to run them against different adapters with different match files. We can also lean on GTest's facilities for skipping tests if necessary but so far its not.

Reply:
`ctest -L "adapter"` was matching too many tests so I changed this label to `adapter-specific` and added the *Test adapter specific* step to the workflow.